src/CodeGen_LLVM.cpp

```diff
@@ -2748,6 +2741,12 @@ void CodeGen_LLVM::visit(const Evaluate *op) {
     value = NULL;
 }
 
+void CodeGen_LLVM::visit(const Return *op) {
+    llvm::Value *value = codegen(op->value);
```

Comment:
My original goal was to enable defining lowered functions that returned the value returned by another Halide function, in fact the motivation for adding Return was to preserve the return code of a Halide pipeline when called from another LoweredFunc: https://github.com/halide/Halide/pull/699/files#diff-c3cda5fa5b7d2d065eb79cb2c4d4ab2eR2091

In general, I think it will be pretty useful. One of the things easily enabled by this branch (and, maybe, the Return IR node) would be to make a LoweredFunc that dispatches to a bunch of other Halide pipelines, e.g. on GeneratorParams, or target architectures. I was picturing it with Return, but maybe this (and the above already existing example) works without Return? 

I think you're right about cleaning up/asserting there are no allocations. I will come back to that.